src/libcollections/btree/node.rs

```diff
@@ -1063,6 +1067,40 @@ impl<'a, K, V> Handle<NodeRef<marker::Mut<'a>, K, V, marker::Internal>, marker::
             Handle::new_edge(self.node, self.idx)
         }
     }
+
+    pub fn steal_left(mut self)
```

Comment:
Could these functions be moved to `map.rs`? The intention was for `node.rs` to act like a generic container that doesn't know anything about how B-Trees work, and stealing methods don't really fit that idea.